YakimS/assignment4/IslandsCounter.java

```diff
@@ -0,0 +1,55 @@
+package com.shaya.CodeUAss4Package;
+
+import java.util.Arrays;
+
+public class IslandsCounter {
+    /**
+     *  Counts the number of islands in a 2-dimensional array,
+     *  which each tile of it is either land(=true) or water(=false).
+     *  Two tiles belong to the same island if they are both land and are
+     *  adjacent horizontally or vertically, but not diagonally.
+     */
+    public int countIsland(boolean[][] boolMap) {
+        int[][] islandNumMap = new int[boolMap.length][boolMap[0].length];

```

Comment:
What if somebody passes `{}`?

Reply:
Right! I've added the relevant test and fix.